test_zeroconf.py

```diff
@@ -807,6 +807,112 @@ def remove_service(self, zeroconf, type, name):
+                service_updated.set()
+
+        listener = MyListener()
+        zeroconf_browser = Zeroconf(interfaces=['127.0.0.1'])
+        zeroconf_browser.add_service_listener(subtype, listener)
+
+        properties = dict(
+            prop_none=None,
+            prop_string=b'a_prop',
+            prop_float=1.0,
+            prop_blank=b'a blanked string',
+            prop_true=1,
+            prop_false=0,
+        )
+
+        zeroconf_registrar = Zeroconf(interfaces=['127.0.0.1'])
+        desc = {'path': '/~paulsm/'}  # type: r.ServicePropertiesType
+        desc.update(properties)
+        info_service = ServiceInfo(
+            subtype, registration_name,
+            socket.inet_aton("10.0.1.2"), 80, 0, 0,
+            desc, "ash-2.local.")
+        zeroconf_registrar.register_service(info_service)
+
+        try:
+            service_added.wait(1)
+            assert service_added.is_set()
+
+            # short pause to allow multicast timers to expire
+            time.sleep(3)
+
+            # clear the answer cache to force query
+            for record in zeroconf_browser.cache.entries():
+                zeroconf_browser.cache.remove(record)
+
+            # get service info without answer cache
+            info = zeroconf_browser.get_service_info(type_, registration_name)
+            assert info is not None
+            assert info.properties[b'prop_none'] is False
+            assert info.properties[b'prop_string'] == properties['prop_string']
+            assert info.properties[b'prop_float'] is False
+            assert info.properties[b'prop_blank'] == properties['prop_blank']
+            assert info.properties[b'prop_true'] is True
+            assert info.properties[b'prop_false'] is False
+
+            info = zeroconf_browser.get_service_info(subtype, registration_name)
+            assert info is not None
+            assert info.properties[b'prop_none'] is False
+
+            # Begin material test addition
+            sublistener = MySubListener()
+            zeroconf_browser.add_service_listener(registration_name, sublistener)
+            properties['prop_blank'] = b'an updated string'
+            desc.update(properties)
+            info_service = ServiceInfo(
+                subtype, registration_name,
+                socket.inet_aton("10.0.1.2"), 80, 0, 0,
+                desc, "ash-2.local.")
+            zeroconf_registrar.update_service(info_service)
+            service_updated.wait(1)
+            assert service_updated.is_set()
+
+            info = zeroconf_browser.get_service_info(type_, registration_name)
+            assert info is not None
+            assert info.properties[b'prop_blank'] == properties['prop_blank']
+            # End material test addition
+
+            zeroconf_registrar.unregister_service(info_service)
+            service_removed.wait(1)
+            assert service_removed.is_set()
+
+        except AssertionError:
+            raise
```

Comment:
This `except` block can be removed, it doesn't seem to be doing anything.

Reply:
Agreed, missed in my review!